network/mesh.go

```diff
@@ -0,0 +1,276 @@
+// Copyright (C) 2019-2025 Algorand, Inc.
+// This file is part of go-algorand
+//
+// go-algorand is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// go-algorand is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with go-algorand.  If not, see <https://www.gnu.org/licenses/>.
+
+package network
+
+import (
+	"context"
+	"errors"
+	"math/rand"
+	"sync"
+	"time"
+
+	"github.com/libp2p/go-libp2p/p2p/discovery/backoff"
+)
+
+const meshThreadInterval = time.Minute
+
+type meshStrategy interface {
+	meshThread()
+	start()
+	stop()
+}
+
+type genericMeshStrategy struct {
+	ctx                context.Context
+	meshUpdateRequests chan meshRequest
+	meshThreadInterval time.Duration
+	wg                 sync.WaitGroup
+	meshConfig
+}
+
+type meshConfig struct {
+	backoff        backoff.BackoffStrategy
+	netMesh        func() bool
+	peerStatReport func()
+	closer         func()
+}
+
+type hybridRelayMeshStrategy struct {
+	genericMeshStrategy
+}
+
+type meshStrategyOption func(*meshConfig)
+
+func withMeshExpJitterBackoff() meshStrategyOption {
```

Comment:
Should the invoking of this be behind an `initP2PMesh` type function housed in this file? Right now, the initialization logic is in `p2pNetwork.go`.